src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -251,7 +251,7 @@ private static VersionInfo GetVersionInfo()
                 loaderAssemblyName: typeof(NormalizerTransformer).Assembly.FullName);
         }
 
-        private class ColumnInfo
+        public class ColumnInfo
```

Comment:
>ColumnInfo [](start = 21, length = 10)

I feel like we ought to have *some* way to access the parameters, but this wholesale exposing of the guts of normalizer is not something I read with great enthusiasm. I do not like this idea especially of exposing `ColumnInfo`.

I recommend having a very, very limited structure parallel to `ColumnInfo`, with one abstract class and three concrete subclasses. (All non-instantiable publicly.) This would expose exactly what we need to expose out of the three types of normalizers, and absolutely nothing else.

Reply:
seems like if we expose just 
- IColumnFunction 
- the : readonly IReadOnlyList<IColumnFunction> ColumnFunctions
-  the interfaces that contain the weights definition we should get it all. Let me know if this is still undesirable, and i can change it. 

---
In reply to: [232432671](https://github.com/dotnet/machinelearning/pull/1594#discussion_r232432671) [](ancestors = 232432671)

Reply:
My second attempt at getting the IColumnFunction and casting them to the various NormalizerData was hacky. It relied on the fact that every implementation down the road implemented both IColumnFunction and the INormalizerTypeData interfaces. Changed it in favor of the abstract GetModelParams returned by the type that actually is an INormalizerTypeData. Creates some repetition, but seems more correct. 

Exposed those all through the parallel ModelParams.

---
In reply to: [232788584](https://github.com/dotnet/machinelearning/pull/1594#discussion_r232788584) [](ancestors = 232788584,232432671)